compiler/rustc_middle/src/query/mod.rs

```diff
@@ -121,6 +122,21 @@ rustc_queries! {
         desc { "perform lints prior to macro expansion" }
     }
 
+    /// Tracked access to environment variables.
+    ///
+    /// Useful for the implementation of `std::env!`, `proc-macro`s change
+    /// detection and other changes in the compiler's behaviour that is easier
+    /// to control with an environment variable than a flag.
+    ///
+    /// NOTE: This currently does not work with dependency info in the
+    /// analysis, codegen and linking passes, place extra code at the top of
+    /// `rustc_interface::passes::write_dep_info` to make that work.
+    query env_var(key: &'tcx OsStr) -> Option<&'tcx OsStr> {
```

Comment:
Now that this returns `OsStr`, should it be called `env_var_os`? And then we can add a helper function on the side that converts into a string if UTF-8?

Reply:
I agree.

Reply:
Have created one now in `rustc_middle/src/util/mod.rs`, though unsure if there's a better place for it.